packages/webapp/src/containers/Finances/AddSale/index.jsx

```diff
@@ -1,110 +1,98 @@
+ *
+ *  LiteFarm is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  LiteFarm is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details, see <https://www.gnu.org/licenses/>.
+ */
+
 import React from 'react';
-import CropSaleForm from '../../../components/Forms/CropSale';
-import GeneralRevenueForm from '../../../components/Forms/GeneralRevenue';
-import { addOrUpdateSale } from '../actions';
-import { userFarmSelector, measurementSelector } from '../../userFarmSlice';
-import { currentAndPlannedManagementPlansSelector } from '../../managementPlanSlice';
+import GeneralRevenue from '../../../components/Forms/GeneralRevenue';
+import useCropSaleInputs from '../useCropSaleInputs';
+import { addSale } from '../actions';
+import { userFarmSelector } from '../../userFarmSlice';
 import { useDispatch, useSelector } from 'react-redux';
 import { useTranslation } from 'react-i18next';
 import { useCurrencySymbol } from '../../hooks/useCurrencySymbol';
 import { hookFormPersistSelector } from '../../hooks/useHookFormPersist/hookFormPersistSlice';
 import { HookFormPersistProvider } from '../../hooks/useHookFormPersist/HookFormPersistProvider';
-import { revenueTypeSelector } from '../../revenueTypeSlice';
-import { getRevenueFormType } from '../util';
-import { revenueFormTypes as formTypes } from '../constants';
+import { revenueTypeByIdSelector, revenueTypeTileContentsSelector } from '../../revenueTypeSlice';
 
 function AddSale() {
-  const { t } = useTranslation();
+  const { t } = useTranslation(['translation', 'revenue']);
   const dispatch = useDispatch();
 
-  const managementPlans = useSelector(currentAndPlannedManagementPlansSelector) || [];
   const farm = useSelector(userFarmSelector);
-  const system = useSelector(measurementSelector);
   const persistedFormData = useSelector(hookFormPersistSelector);
   const { revenue_type_id } = persistedFormData || {};
-  const revenueType = useSelector(revenueTypeSelector(revenue_type_id));
+  const revenueType = useSelector(revenueTypeByIdSelector(revenue_type_id));
+  const revenueTypes = useSelector(revenueTypeTileContentsSelector);
+
+  const revenueTypeReactSelectOptions = revenueTypes?.map((type) => {
+    const retireSuffix = type.deleted ? ` ${t('REVENUE.EDIT_REVENUE.RETIRED')}` : '';
 
-  const formType = getRevenueFormType(revenueType);
+    return {
+      value: type.revenue_type_id,
+      label: type.farm_id
+        ? type.revenue_name + retireSuffix
+        : t(`revenue:${type.revenue_translation_key}`),
+    };
+  });
 
   const onSubmit = (data) => {
-    const addSale = {
+    const editedSale = {
```

Comment:
Since we're not editing here, maybe we could call this `newSale` or something like that?

Reply:
Copy mistake thank you!